org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/OpenCallHierarchyHyperlinkDetector.java

```diff
@@ -36,7 +36,7 @@ public class OpenCallHierarchyHyperlinkDetector extends JavaElementHyperlinkDete
 
 	@Override
 	protected void addHyperlinks(List<IHyperlink> hyperlinksCollector, IRegion wordRegion, SelectionDispatchAction openAction, IJavaElement element, boolean qualify, JavaEditor editor) {
-		if ((element.getElementType() == IJavaElement.METHOD || element.getElementType() == IJavaElement.FIELD) && (element instanceof IMember imember)) {
+		if (CallHierarchy.isPossibleInputElement(element) && (element instanceof IMember imember)) {
```

Comment:
The second part of condition seem to be unnecessary, see
https://github.com/eclipse-jdt/eclipse.jdt.ui/blob/2630820c16c8887ad8e20fa0a71cf9bd61ee0eba/org.eclipse.jdt.core.manipulation/core%20extension/org/eclipse/jdt/internal/corext/callhierarchy/CallHierarchyCore.java#L305

Reply:
Yeah, I was aware of that. I left it unchanged to avoid explicit casting afterwards ... which I can imagine being seen as potential cause of ClassCastException to somebody looking just at this piece of code. If that's not of any concern, then I agree the check is unnecessary.

Reply:
OK, I see.